docs/design-proposals/multi-tenancy/multi-tenancy-network.md

```diff
@@ -120,10 +128,118 @@ The reasons that DNS service is per-network instead of per-space are:
 * VPC networks are isolated by nature. It will be difficult to have a shared DNS pod that's accessible to pods in various VPC networks.
 
 ### Service
-For VPC-isolated networking, service IP support has to be provided w/o relying on iptable/IPVS rules on host netns. In other words, traditional kube-proxy can not used in VPC-isolated networking.
 
+#### Type definition
+
+Addition of service spec is metadata.network field. By default, it is the default network of the tenant. 
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  network: my-network
+  selector:
+    app: MyApp
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+```
+
+#### Naming Problem & Solution
+
+Services, as one of the critical network related resource types, are withing the network boundary. There is no such *shared* services of tenant or whole cluster, as both the critical virtual IP address and the associated endpoints (typically leading to pods) are all of same network.
+
+Putting service type in tenant level is out of primacy as it has following flaws:
+1. service IP address has to be the same across all networks, which implies some range of service IP needs to be predefined as shared across networks of a tenant (tenant manageability burden);  
+1. associating network-agnostic service to network-specific endpoints would introduce significant inconsistency;
+1. kubectl query for service (e.g. ```kubectl describe service```) has no way to return meaningful content of endpoints.
+
+The well-known services (default/kubernetes, kube-system/kube-dns) present some problem here. Service kubernetes has tenant/namespace(default) scope in current implementation, unable to have multiple kubernets services with each of them in one network due to name conflict. The same rational applies to kube-dns.
```

Comment:
I don't fully understand the problem here.
1. What is the scope in current implementation?
1. Why placing services in namespaces is not sufficient?

Reply:
This is the very tricky situation. Taking the "kubernetes" service: in multi-tenancy arktos, each tenant would have its own kubernetes service in the default namespace; we cannot put another same-named "kubernetes" service in the same "deafult" namespace - kubernetes apiserver just won't allow it. What we need is EACH network should has one "kubernetes" service, scoped in "default" namespace, and serves for the specific network only.